include/guttering_system.h

```diff
@@ -1,38 +1,51 @@
 #pragma once
 #include "types.h"
+#include "work_queue.h"
 #include <math.h>
 
 class GutteringSystem {
 public:
-  virtual insert_ret_t insert(const update_t &upd) = 0; // insert an element to the buffering system
-  virtual bool get_data(data_ret_t& data) = 0;          // get data out of the buffering system
-  virtual flush_ret_t force_flush() = 0;                // force all data out of buffers
-  virtual void set_non_block(bool block) = 0;           // set non-blocking calls in the work queue
-
+  GutteringSystem(node_id_t num_nodes, int workers, bool page_slots=false) : 
+   leaf_gutter_size((configure_system(), 
+      std::max((node_id_t) (gutter_factor * sketch_size(num_nodes)), (node_id_t)1))),
+   wq(workers * queue_factor,
+      page_slots ? leaf_gutter_size + page_size / sizeof(node_id_t) : leaf_gutter_size) {}
+  
   virtual ~GutteringSystem() {};
+  virtual insert_ret_t insert(const update_t &upd) = 0; //insert an element to the guttering system
+  virtual flush_ret_t force_flush() = 0;                //force all data out of buffers
 
   // get the size of a work queue elmement in bytes
   int gutter_size() {
-    return upds_per_gutter() * sizeof(size_t);
+    return leaf_gutter_size * sizeof(node_id_t);
   }
 
-  static double sketch_size(node_id_t num_nodes) {
-    return 42 * sizeof(node_id_t) * pow(log2(num_nodes), 2) / (log2(3) - 1); 
+  // returns the number of node_id_t types that fit in a sketch
+  static node_id_t sketch_size(node_id_t num_nodes) {
```

Comment:
Why is the return type node_it_t instead of, say, size_t? We may also want to rename sketch_size to something like sketches_per since the unit of the result changed (from bytes to counts).

Reply:
Both of these have been changed.